base/applications/utilman/process.c

```diff
@@ -147,43 +185,28 @@ BOOL LaunchProcess(LPCWSTR lpProcessName)
  */
 BOOL CloseProcess(IN LPCWSTR lpProcessName)
 {
-    BOOL bSuccess = FALSE;
-    PROCESSENTRY32W Process = {0};
+    HANDLE hProcess;
+    DWORD dwProcessID;
 
-    /* Create a snapshot and check if the given process' executable name is running */
-    HANDLE hSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
-
-    if (hSnapshot == INVALID_HANDLE_VALUE)
+    /* Get the process ID */
+    dwProcessID = GetProcessID(lpProcessName);
+    if (dwProcessID == 0)
+    {
         return FALSE;
+    }
 
-    Process.dwSize = sizeof(Process);
+    /* Open the process so that we can terminate it */
+    hProcess = OpenProcess(PROCESS_TERMINATE, 0, dwProcessID);
 
-    /* Enumerate the processes */
-    if (Process32FirstW(hSnapshot, &Process))
+    /* Make sure that we do not terminate ourselves of course */
+    if ((hProcess != NULL) && (dwProcessID != GetCurrentProcessId()))
```

Comment:
Maybe add the `dwProcessID != GetCurrentProcessId()` check to the condition above? You'd still return FALSE, but avoid the unnecessary process open.